### DISCUSSION TITLE
BRAM model "input EN" behavior question

### DISCUSSION THREAD

**Discussion by yukaii2019** [bol-edu/HLS-SOC-Discussions#105] [Q&A]
大家好：

想問BRAM中read的部分的timing問題。照workbook的波形EN與ADDR應該在同個cycle被sample，但助教給的behavior model是ADDR先然後下個cycle要EN為1才能讀到output，並沒有在同個cycle，想請問哪種才是正確的。謝謝。

**Comment by boledulab** [comment]
EN will gate both input and output.
